### ISSUE TITLE
[Bug]: sqlParser can not parse Function/Procedure ddl when DEFINER=`root`@`%` is in ddl in mysql mode

### ISSUE THREAD

**Issue by PeachThinking** [oceanbase/odc#838]
### ODC version

4.2.2

### OB version

mysql 5.7

### What happened?

PLParser of obmysql can not pase function ddl like:
```SQL
CREATE DEFINER=`root`@`%` FUNCTION `test`(
	p1 VARCHAR(45)) RETURNS int(11)
    DETERMINISTIC
begin
	return 1;
end
```
because `root`@`%` can not match PLLexer

### What did you expect to happen?

pasrse ddl like above success

### How can we reproduce it (as minimally and precisely as possible)?

in mysql mode, get ddl by query "show create function" and use PLParser to parse the ddl

**Comment by yhilmare** [comment]
has been fixed #837

**Comment by tonghan-gsh** [comment]
![image](https://github.com/oceanbase/odc/assets/46598871/b8fe249b-0244-4771-8f5b-1b8c3b8bf25f)
pass
